src/verify.rs

```diff
@@ -446,13 +420,10 @@ pub struct VerifyResult {
 
 impl VerifyResult {
     pub fn diagnostics(&self) -> Vec<(StatementAddress, Diagnostic)> {
-        let mut out = Vec::new();
-        for vsr in self.segments.values() {
-            for (&sa, &ref diag) in &vsr.diagnostics {
-                out.push((sa, diag.clone()));
-            }
-        }
-        out
+        self.segments
+            .values()
+            .flat_map(|vsr| vsr.diagnostics.iter().map(|(&sa, diag)| (sa, diag.clone())))
```

Comment:
I don't think this is a readability improvement.  There was a lot of code like this, but the borrow checker hated it, and eventually I realized it was mostly inscrutable even to me, so I rewrote it with loops and have been much happier since.